libp2p/protocols/pubsub/rpc/message.nim

```diff
@@ -66,17 +66,16 @@ proc init*(
   var msg = Message(data: data, topicIDs: @[topic])
 
   # order matters, we want to include seqno in the signature
-  if seqno.isSome:
-    msg.seqno = @(seqno.get().toBytesBE())
+  seqno.withValue(seqn):
+    msg.seqno = @(seqn.toBytesBE())
 
-  if peer.isSome:
-    let peer = peer.get()
+  peer.withValue(peer):
     msg.fromPeer = peer.peerId
     if sign:
       msg.signature = sign(msg, peer.privateKey).expect("Couldn't sign message!")
       msg.key = peer.privateKey.getPublicKey().expect("Invalid private key!")
         .getBytes().expect("Couldn't get public key bytes!")
-  elif sign:
+  if peer.isNone() and sign:
```

Comment:
peer is Option

Reply:
@lchenut ?

Reply:
And seqno aswell, yes, but they're arguments of the proc, and the proc is exposed. So it's a breaking change.